partner_contact_in_several_companies/models/res_partner.py

```diff
@@ -0,0 +1,158 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    contact_type = fields.Selection(
+        [
+            ("standalone", "Standalone Contact"),
+            ("attached", "Attached to existing Contact"),
+        ],
+        compute="_compute_contact_type",
+        store=True,
+        index=True,
+        default="standalone",
+    )
+    contact_id = fields.Many2one(
+        "res.partner",
+        string="Main Contact",
+        domain=[("is_company", "=", False), ("contact_type", "=", "standalone")],
+    )
+    other_contact_ids = fields.One2many(
+        "res.partner",
+        "contact_id",
+        string="Others Positions",
+    )
+
+    @api.depends("contact_id")
+    def _compute_contact_type(self):
+        for rec in self:
+            rec.contact_type = "attached" if rec.contact_id else "standalone"
+
+    def _basecontact_check_context(self, mode):
+        """Remove "search_show_all_positions" for non-search mode.
+        Keeping it in context can result in unexpected behaviour (ex: reading
+        one2many might return wrong result - i.e with "attached contact"
+        removed even if it"s directly linked to a company).
+        Actually, is easier to override a dictionary value to indicate it
+        should be ignored...
+        """
+        if mode != "search" and "search_show_all_positions" in self.env.context:
+            result = self.with_context(search_show_all_positions={"is_set": False})
+        else:
+            result = self
+        return result
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        """Display only standalone contact matching ``args`` or having
+        attached contact matching ``args``"""
+        ctx = self.env.context
+        if (
```

Comment:
In condition will may be raised exception KeyError.
Add the variable:
```
searsearch_show_all_positions = ctx.get("search_show_all_positions", {})
```
And then:
```
if searsearch_show_all_positions.get("is"set") and not searsearch_show_all_positions.get("set_value"):
```

Reply:
Thanks for the example, I will fix it